manticore/platforms/evm.py

```diff
@@ -2103,6 +2154,12 @@ def _process_pending_transaction(self):
         if failed:
             self._close_transaction('TXERROR', rollback=True)
 
+        if caller not in self._world_state:
+            self._world_state[caller] = { 'nonce' : 1 }
+        elif sort == 'CREATE' or not self._world_state[caller]['code']:
```

Comment:
I think you may want to move this to _open_transaction https://github.com/trailofbits/manticore/blob/2a4d62a2cc634a27c452841f420eb4e329532c32/manticore/platforms/evm.py#L1780

We need to do something to save the old nonce for the caller. Probably here https://github.com/trailofbits/manticore/blob/2a4d62a2cc634a27c452841f420eb4e329532c32/manticore/platforms/evm.py#L1803

So we can then rollback if the creation failed/reverted/throwed ?
https://github.com/trailofbits/manticore/blob/2a4d62a2cc634a27c452841f420eb4e329532c32/manticore/platforms/evm.py#L1815

Reply:
This is resolved by commit 25a2a2a307bb9cbd0e37582bd8fbdf46053cc377